From 29555e143d01de7570bdb5d11d9c785f6e59e6bf Mon Sep 17 00:00:00 2001
From: Ayala Beker <ayala.beker@intel.com>
Date: Thu, 14 Jul 2022 09:33:36 +0300
Subject: [PATCH] iw: scan: set NL80211_SCAN_FLAG_COLOCATED_6GHZ in case of
 full sched scan

Same as in regular scan, in case of full sched scan need to set
NL80211_SCAN_FLAG_COLOCATED_6GHZ flag, to scan for collocated
APs by default.

Signed-off-by: Ayala Beker <ayala.beker@intel.com>
Signed-off-by: Johannes Berg <johannes.berg@intel.com>
---
 scan.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scan.c b/scan.c
index dfc136a..4c67c87 100644
--- a/scan.c
+++ b/scan.c
@@ -349,6 +349,8 @@ int parse_sched_scan(struct nl_msg *msg, int *argc, char ***argv)
 		nla_put_nested(msg, NL80211_ATTR_SCAN_SSIDS, ssids);
 	if (have_freqs)
 		nla_put_nested(msg, NL80211_ATTR_SCAN_FREQUENCIES, freqs);
+	else
+		flags |= NL80211_SCAN_FLAG_COLOCATED_6GHZ;
 	if (have_matchset)
 		nla_put_nested(msg, NL80211_ATTR_SCHED_SCAN_MATCH, matchset);
 	if (have_plans)
-- 
2.41.0.694.ge786442a9b-goog

